### DISCUSSION TITLE
Debugging casters

### DISCUSSION THREAD

**Discussion by ju5t** [livewire/livewire#4885] [1. Help] [answered]
One of our components shows the exchange rate between two currencies. A scheduled job, that runs overnight, caches the result in a DTO. We use a DTO as we need to store an immutable Carbon object along with some other details of the current rate. Our DTO's are based on Spatie's DTO package.

I know Livewire only supports a limited list of data types because JS needs to process them too.

From the documentation I understand I needed a caster for the DTO. I made one. Pretty generic and simple. But it's not working. Probably because I did something wrong. It still errors on the public property type not being supported. 

I have added it to a protected property `$casts`, but I think it's not read. I did some source diving but couldn't find how Livewire knows what casts to use. I understand that code helps, but for now I'm most curious about the debugging part.

What is the best approach to debug the caster?

**Comment by joshhanley** [comment, marked as answer]
@ju5t `$casts` were removed in V2, see the upgrade guide here https://laravel-livewire.com/docs/2.x/upgrading#casters

What you want is to add the `wireable` interface to the DTO, see here https://laravel-livewire.com/docs/2.x/properties#wireable-properties

Hope this helps!

**Reply by ju5t** [reply]
Thanks! I really thought I read about `$casts` in the docs, but apparently I didn't :)
